src/main/java/uk/gov/companieshouse/insolvency/data/service/InsolvencyServiceImpl.java

```diff
@@ -114,7 +115,19 @@ public CompanyInsolvency retrieveCompanyInsolvency(String companyNumber) {
     }
 
     @Override
-    public void deleteInsolvency(String contextId, String companyNumber) {
+    public void deleteInsolvency(String contextId, String companyNumber)
+            throws EmptyResultDataAccessException {
+        Optional<InsolvencyDocument> insolvencyDocumentOptional =
+                insolvencyRepository.findById(companyNumber);
+
+        if (insolvencyDocumentOptional.isEmpty()) {
+            logger.info(String.format(
+                    "Company insolvency doesn't exist for company number %s",
+                    companyNumber));
+            throw new EmptyResultDataAccessException(0);
+        }
+
+        insolvencyRepository.deleteById(companyNumber);
```

Comment:
I think it should be wrapped in a try catch as the above - `findById` and `deleteById` could fail if a connection issue occurs 

```
} catch (DataAccessException dbException) {
            throw new ServiceUnavailableException(dbException.getMessage());
        }
```